core/profile.c

```diff
@@ -1236,15 +1236,15 @@ static void calculate_gas_information_new(const struct dive *dive, const struct
 	enum divemode_t current_divemode = UNDEF_COMP_TYPE;
 
 	for (i = 1; i < pi->nr; i++) {
-		int fn2, fhe;
+		double fn2, fhe;
 		struct plot_data *entry = pi->entry + i;
 
 		gasmix = get_gasmix(dive, dc, entry->sec, &evg, gasmix);
 		amb_pressure = depth_to_bar(entry->depth, dive);
 		current_divemode = get_current_divemode(dc, entry->sec, &evd, &current_divemode);
 		fill_pressures(&entry->pressures, amb_pressure, gasmix, (current_divemode == OC) ? 0.0 : entry->o2pressure.mbar / 1000.0, current_divemode);
-		fn2 = (int)(1000.0 * entry->pressures.n2 / amb_pressure);
-		fhe = (int)(1000.0 * entry->pressures.he / amb_pressure);
+		fn2 = (1000.0 * entry->pressures.n2 / amb_pressure);
+		fhe = (1000.0 * entry->pressures.he / amb_pressure);
```

Comment:
Small style comment: You could remove the parentheses in the previous two lines.

No comment on the rest - this is above my pay-grade. Turning these two into floating points looks reasonable though.

Reply:
you still could drop those parenthesis, right?